api/test/test_edr.py

```diff
@@ -0,0 +1,186 @@
+            0
+        ].temporal_interval.start.ToDatetime().strftime("%Y-%m-%d %H:%M:%S")
+        assert "2022-12-31 01:00:01" == mock_getObsRequest.call_args[0][0].temporal_interval.end.ToDatetime().strftime(
+            "%Y-%m-%d %H:%M:%S"
+        )
+
+        assert response.status_code == 200
+        assert response.json() == compare_data
+
+
+def test_get_area_with_incorrect_coords():
+    response = client.get("/collections/observations/area?coords=POLYGON((22.12 59.86, 24.39 60.41))")
+
+    assert response.status_code == 422
+
+
+def test_get_area_with_incorrect_geometry_type():
+    response = client.get("/collections/observations/area?coords=POINT(22.12 59.86)")
+
+    assert response.status_code == 422
+
+
+def test_get_position_with_normal_query():
+    # Wrap the original get_data_area to a mock so we can assert against the call values
+    with patch("routers.edr.get_data_area", wraps=edr.get_data_area) as mock_get_data_area, patch(
+        "routers.edr.getObsRequest"
+    ) as mock_getObsRequest:
+        test_data = load_json("test/test_data/test_coverages_proto.json")
+        compare_data = load_json("test/test_data/test_coverages_covjson.json")
+
+        mock_getObsRequest.return_value = create_mock_obs_response(test_data)
+
+        response = client.get(
+            "/collections/observations/position?coords=POINT(5.179705 52.0988218)"
+            "&parameter-name=TA_P1D_AVG&datetime=2022-12-31T00:00Z/2022-12-31T00:00Z"
+        )
+
+        mock_get_data_area.assert_called_once()
+        mock_get_data_area.assert_called_with(
+            "POLYGON ((5.179805 52.0988218, 5.179705 52.0987218, 5.1796050000000005 52.0988218, "
+            "5.179705 52.09892180000001, 5.179805 52.0988218))",
+            "TA_P1D_AVG",
+            "2022-12-31T00:00Z/2022-12-31T00:00Z",
+            "covjson",
+        )
+
+        assert response.status_code == 200
+        assert response.json() == compare_data
+
+
+def test_get_position_with_incorrect_coords():
+    response = client.get("/collections/observations/position?coords=POINT(60.41)")
+
+    assert response.status_code == 422
+    assert response.json() == {"detail": {"coords": "Invalid coordinates: POINT(60.41)"}}
+
+
+def test_get_position_with_incorrect_geometry_type():
+    response = client.get(
+        "/collections/observations/position?coords=POLYGON((22.12 59.86, 24.39 60.41, "
+        "24.39 60.41, 24.39 59.86, 22.12 59.86))"
+    )
+
+    assert response.status_code == 422
+    assert response.json() == {
+        "detail": {
+            "coords": "Invalid coordinates: POLYGON((22.12 59.86, 24.39 60.41, 24.39 60.41, 24.39 59.86, 22.12 59.86))"
+        }
+    }
+
+
+def create_mock_obs_response(json_data):
+    response = dstore.GetObsResponse()
+    Parse(json.dumps(json_data), response)
+    return response
+
+
+def load_json(file_path):
+    with open(file_path, "r") as file:
+        return json.load(file)
```

Comment:
Duplicate with other test file?

Reply:
Added utilities.py to test folder containing helper functions. https://github.com/EURODEO/e-soh/pull/30/commits/b34ed81668c048efaf6160a9fa8df2daf8eb1597